lib/account.rb

```diff
@@ -0,0 +1,106 @@
+require 'yaml'
+require 'pry'
+
+class Account
+  include BankErrors
+  include Validation
+  include FileLoader
+  include MoneyOperations
+
+  attr_reader :name, :age, :login, :password, :errors, :current_account, :card, :file_path
+
+  CARD_TYPES = {
+    usual: UsualCard,
+    capitalist: CapitalistCard,
+    virtual: VirtualCard
+  }.freeze
+
+  LOGIN_LENGTH = (4..20).freeze
+  PASSWORD_LENGTH = (6..30).freeze
+  AGE_RANGE = (23..90).freeze
+
+  def initialize
+    @errors = []
+    @file_path = 'accounts.yml'
+
+    @name = nil
+    @login = nil
+    @age = 0
+    @password = nil
+    @card = []
+    @current_account = nil
+  end
+
+  def credentials(name:, age:, login:, password:)
+    name_errors(name: name, errors: @errors)
+    age_errors(age: age, errors: @errors, range: AGE_RANGE)
+    login_errors(login: login, errors: @errors, length: LOGIN_LENGTH)
+    password_errors(password: password, errors: @errors, length: PASSWORD_LENGTH)
+
+    @name = name
+    @age = age
+    @login = login
+    @password = password
+  end
+
+  def create
+    new_accounts = accounts << @current_account = self
+    save_in_file(new_accounts, @file_path)
+  end
+
+  def load(login, password)
+    @current_account = accounts.find { |a| login == a.login && password == a.password }
+
+    raise_error(NoAccountError) if @current_account.nil?
+  end
```

Comment:
Maybe will be better to create separate service for functions which work with yaml?

Reply:
Yep, I have one -- lib/modules/file_loader.rb -- which works directly with file operations. Other classes that need any info from file, use file_loader methods.